### DISCUSSION TITLE
Can't use contexts in IDE. Is it implemented or not?

### DISCUSSION THREAD

**Discussion by adjenks** [TabbyML/tabby#3626] [Q&A] [answered]
**Describe the bug**
I thought I could add documents with "@" but if I type @ in the chat it doesn't do anything.

**Information about your version**
Extension is 1.16.0

**Additional context**
Is there a repo for the extension? I couldn't find it. Anyhow, I can't seem to use any of the context documents I've added to my tabby instance in the settings under context providers. Is this feature not implemented yet? I can use it in the web interface fine.

**Comment by wsxiaoys** [comment]
Hi! Currently, the chat side panel only supports code repository context, as document context is not yet supported.

**Comment by adjenks** [comment]
@wsxiaoys
Okay, understood. Perhaps this should be made more clear in the extension.
Is there a repo for the extension? Or is that particular code closed source?

**Reply by wsxiaoys** [reply]
Great use case for tabby's Answer Engine:

https://demo.tabbyml.com/search/tabby-ml-tabby-where-does-vscode-extension-code-l-GJrOvG

**Comment by adjenks** [comment, marked as answer]
Your link didn't work the first time I clicked on it, it reset the page. So I asked myself. It said this:

>  Where are the repos for the tabbyml vscode and vs-codium extensions?


> Answer
> 
> The repositories for the tabbyml VSCode and VS-Codium extensions are hosted on GitHub. Here are the links to the repositories:
> 
>     tabbyml VSCode Extension:
>         Repository: [tabby-vscode](https://github.com/TabbyML/tabby-vscode)
>         This extension integrates TabbyML's AI-powered code completion into Visual Studio Code.
> 
>     tabbyml VS-Codium Extension:
>         Repository: [tabby-vscodium](https://github.com/TabbyML/tabby-vscodium)
>         This extension is tailored for VS-Codium, a free/libre open-source version of Visual Studio Code.
> 
> These repositories contain the source code, documentation, and instructions for installing and using the extensions. If you have any questions or need further assistance, feel free to ask! 😊

This answer seems to be a hallucination. So I clicked your link again and it worked the second time... I see now. Thanks I suppose.

In yours it says: 

> 
> The VSCode extension code for TabbyML/tabby is located in the clients/vscode directory of the repository. This directory contains the source code, configuration, and related files for the VSCode extension 11
> . For example, the main extension entry point is found in clients/vscode/src/extension.ts, which handles the activation and setup of the extension 12
> . Additionally, the clients/vscode/CONTRIBUTING.md file provides guidance on how to set up and develop the extension locally 11
> .

I think it worked because you included the repo. My error was in assuming that it was in an entirely separate repo.
